src/engine/input-controllers/basic-keyboard-movement.ts

```diff
@@ -0,0 +1,36 @@
+import KeyboardMapping from './keyboard-mapping'
+import VelocityModule from '../modules/velocity-module'
+
+export class BasicKeyboardMovementMapping extends KeyboardMapping {
+  velocityModule: VelocityModule;
+
+  constructor(velocityModule: VelocityModule, keys: Keyset) {
+    super();
+
+    this.velocityModule = velocityModule;
+
+    super.registerOnDown(keys.left, () => this.velocityModule.accelerateLeft());
+    super.registerOnDown(keys.right, () => this.velocityModule.accelerateRight());
+    super.registerOnDown(keys.up, () => this.velocityModule.accelerateUp());
+    super.registerOnDown(keys.down, () => this.velocityModule.accelerateDown());
+
+    super.registerOnUp(keys.left, () => this.velocityModule.decelerateLeft());
+    super.registerOnUp(keys.right, () => this.velocityModule.decelerateRight());
+    super.registerOnUp(keys.up, () => this.velocityModule.decelerateUp());
+    super.registerOnUp(keys.down, () => this.velocityModule.decelerateDown());
```

Comment:
Shouldn't these be `this._____` instead of `super._____`? Right now, you haven't overridden the functions so `this` should act the same, and will avoid bugs if you do override them.